src/main/java/org/codeseoul/event_member_management/member/MemberRepository.java

```diff
@@ -1,6 +1,11 @@
 package org.codeseoul.event_member_management.member;
 
+import java.util.Optional;
+
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface MemberRepository extends JpaRepository<Member, Long> {
+    public Optional<Member> findByUsername(String username);
+    public Optional<Member> findByPhoneNumber(String phoneNumber);
+    public Optional<Member> findByEmail(String email);
```

Comment:
As mentioned elsewhere, I think you can replace with `findByUsernameOrPhoneNumberOrEmail`. I know SQL `OR` isn't the most performant, but I think the data volume will be low enough at the beginning that it won't be worth worrying about.

Reply:
I have an idea, but I'm not sure what the SQL would be. The repository could just return a list of strings that correlate to any existing username, email, and password. Then, we would just need to check each changed value and return an error if it's on the list.

Reply:
It would probably be easier to just pull in all the members and do a programmatic query. That would be more extensible as the jpa method would become unwieldy if you were to add other unique fields later.

Reply:
You want to offload as much set/search logic to the database as you can. The database is usually faster than imperative operations, unless you need to do sequential steps. Even if the method becomes quite large, it's still cleaner than writing the SQL manually and more performant than doing the set filtering at the application layer.